lib/titiler-pgstac-api/runtime/src/handler.py

```diff
@@ -0,0 +1,11 @@
+"""
+Handler for AWS Lambda.
+"""
+
+from mangum import Mangum
+from config import ApiSettings
+from titiler.pgstac.main import app
+
+settings = ApiSettings()
+settings.set_postgres_settings()
```

Comment:
This is kinda useless, we just need to update the environment variables for the `PostgresSettings`
https://github.com/stac-utils/titiler-pgstac/blob/main/titiler/pgstac/settings.py#L36-L51

```python
"""
Handler for AWS Lambda.
"""

import os
from mangum import Mangum
from config import get_secret_dict
from titiler.pgstac.main import app

pgstac_secret_arn = os.environ["PGSTAC_SECRET_ARN"]

secret = get_secret_dict(pgstac_secret_arn)
os.environ.update(
    {
        "postgres_host": secret["host"],
        "postgres_dbname": secret["dbname"],
        "postgres_user": secret["username"],
        "postgres_pass": secret["password"],
        "postgres_port": str(secret["port"]),
    }
)

handler = Mangum(app)
```

Reply:
I find this a bit `hacky`, I think we will be better served with a proper `custom` application! 

```python

import os
from mangum import Mangum

from titiler.pgstac.main import app
from titiler.pgstac.settings import PostgresSettings

import boto3


def get_secret_dict(secret_name: str):
    """Retrieve secrets from AWS Secrets Manager
    Args:
        secret_name (str): name of aws secrets manager secret containing database connection secrets
        profile_name (str, optional): optional name of aws profile for use in debugger only
    Returns:
        secrets (dict): decrypted secrets in dict
    """

    # Create a Secrets Manager client
    session = boto3.session.Session()
    client = session.client(service_name="secretsmanager")

    get_secret_value_response = client.get_secret_value(SecretId=secret_name)

    if "SecretString" in get_secret_value_response:
        return json.loads(get_secret_value_response["SecretString"])
    else:
        return json.loads(base64.b64decode(get_secret_value_response["SecretBinary"]))


@app.on_event("startup")
async def startup_event() -> None:
    """Connect to database on startup."""
    
    pgstac_secret_arn = os.environ["PGSTAC_SECRET_ARN"]
    secret = get_secret_dict(pgstac_secret_arn)
    
    await connect_to_db(
        app,
        settings=PostgresSettings(
            postgres_user: secret["username"],
            postgres_pass: secret["password"],
            postgres_host: secret["host"],
            postgres_port: str(secret["port"]),
            postgres_dbname: secret["dbname"],
        )
    )


handler = Mangum(app)
```

Reply:
One more comment! any reason why not set those variable in the environment directly? https://github.com/developmentseed/eoAPI/blob/main/infrastructure/aws/cdk/app.py#L196-L212

Reply:
🤔 the `build_app` approach seems interesting

Reply:
I simplified following your first comment @vincentsarago.

I think what you describe in your second code suggestion would be ideal, but since it's not critical and would require me re-testing this in deployment before merging, I will stick with the current approach (set up the env in handler.py) and we can always improve things later.